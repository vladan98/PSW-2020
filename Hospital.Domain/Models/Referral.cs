using Hospital.Domain.Enums;
using Hospital.Domain.Interfaces;
using Hospital.Domain.Models.Users;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Hospital.Domain.Models
{
    public class Referral : IIdentifiable<int>
    {
        public int Id { get; set; }
        public Specialization Specialization { get; set; }
        public bool Used { get; set; }
        public int DoctorId { get; set; }
        [NotMapped]
        public virtual Doctor Doctor { get; set; }
        public int PatientId { get; set; }
        [NotMapped]
        public virtual Patient Patient { get; set; }

        public int GetId()
        {
            return Id;
        }
    }
}
