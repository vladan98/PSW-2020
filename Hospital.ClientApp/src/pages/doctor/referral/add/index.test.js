import Referral from "."
import render from "../../../../helpers/renderComponent"

describe("Referral", () => {
    it("should referral form", () => {
        const { container } = render(<Referral />)

        expect(container).toMatchSnapshot()
    })
})