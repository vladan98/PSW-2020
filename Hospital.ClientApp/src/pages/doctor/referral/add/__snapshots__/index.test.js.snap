// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Referral should referral form 1`] = `
<div>
  <main
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="makeStyles-paper-1"
    >
      <form>
        <div
          align="center"
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <div
              class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
        >
          <div
            class="MuiGrid-root MuiGrid-item"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained"
              data-cy="add-referral-submit"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Asign Referral
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </form>
    </div>
  </main>
</div>
`;
